plugin/templates/resource_name.py

```diff
@@ -83,16 +85,21 @@ def __init__(self, collection_config, java_package, oneof):
             self.parent_interface = 'ResourceName'
             self.extension_keyword = 'implements'
         self.parameter_list = [{
-            'parameter': casing_utils.lower_underscore_to_lower_camel(lit),
+            'parameter': symbol_table.getNewSymbol(
+                casing_utils.lower_underscore_to_lower_camel(lit)),
             'parameter_name': lit,
             'not_first': True,
             'not_last': True,
         } for lit in id_segments]
         self.parameter_list[0]['not_first'] = False
         self.parameter_list[-1]['not_last'] = False
         self.format_fields = [{
-            'upper': casing_utils.lower_camel_to_upper_camel(f['parameter']),
+            'upper': casing_utils.lower_underscore_to_upper_camel(
+                f['parameter_name']),
             'lower': f['parameter'],
+            'parameter_name_in_map':
```

Comment:
This is needed in the fieldMap, and the value used in the fieldMap used to just be `lower`, which used to mean `casing_utils.lower_underscore_to_lower_camel(lit)`, but since `lower` has changed, we have to make a new variable that refers back to the old meaning.